bench/wtperf/misc.c

```diff
@@ -125,14 +125,23 @@ backup_read(WT_SESSION *wt_session, const char *from)
     session = (WT_SESSION_IMPL *)wt_session;
 
     /* Open the file handle. */
-    WT_ERR(__wt_open(session, from, WT_FS_OPEN_FILE_TYPE_REGULAR, 0, &fh));
+    if ((ret = __wt_open(session, from, WT_FS_OPEN_FILE_TYPE_REGULAR, 0, &fh)) != 0) {
+        lprintf(wtperf, ret, 0, "Open file handle %s for backup failed", from);
+        goto err;
+    }
     buf = dmalloc(WT_BACKUP_COPY_SIZE);
 
     /* Get the file's size, then read the bytes. */
-    WT_ERR(__wt_filesize(session, fh, &size));
+    if ((ret = __wt_filesize(session, fh, &size)) != 0) {
+        lprintf(wtperf, ret, 0, "Grab file size for %s failed", from);
+        goto err;
+    }
     for (offset = 0; size > 0; size -= n, offset += n) {
         n = WT_MIN(size, WT_BACKUP_COPY_SIZE);
-        WT_ERR(__wt_read(session, fh, offset, (size_t)n, buf));
+        if ((ret = __wt_read(session, fh, offset, (size_t)n, buf)) != 0) {
+            lprintf(wtperf, ret, 0, "Read file %s failed", from);
+            goto err;
+        }
```

Comment:
While it isn't harmful to use the WT internal functions for this, for an application, there is no need to do so. All the same things could be accomplished with system calls of `open(2), stat(2), read(2)`. I don't have strong feelings either way.

Reply:
I have converted to using open, stat, read. I originally thought that open, stat, read was not windows compatible, my mistake